frontend/src/components/search/SaveSearchSelector.tsx

```diff
@@ -0,0 +1,156 @@
+
+export const SaveSearchSelector = ({
+  newSavedSearches,
+  savedSearches,
+  setSavedSearches,
+}: {
+  newSavedSearches: string[];
+  savedSearches: SavedSearchRecord[];
+  setSavedSearches: Dispatch<SetStateAction<SavedSearchRecord[]>>;
+}) => {
+  const t = useTranslations("Search.saveSearch");
+  const { user } = useUser();
+  const { searchParams, replaceQueryParams } = useSearchParamUpdater();
+  const prevSearchParams = usePrevious(searchParams);
+
+  const [selectedSavedSearch, setSelectedSavedSearch] = useState<string>();
+  const [loading, setLoading] = useState<boolean>(false);
+  const [apiError, setApiError] = useState<Error | null>();
+
+  // allows us to avoid resetting the select option when params change after selection
+  const [applyingSavedSearch, setApplyingSavedSearch] =
+    useState<boolean>(false);
+
+  const handleFetchError = useCallback((e: Error) => {
+    setLoading(false);
+    console.error("Error fetching saved searches", e);
+    setApiError(e);
+  }, []);
+
+  const fetchSavedSearches = useCallback(() => {
+    setLoading(true);
+    setApiError(null);
+    return obtainSavedSearches(user?.token)
+      .then((savedSearches) => {
+        setLoading(false);
+        setSavedSearches(savedSearches);
+      })
+      .catch(handleFetchError);
+  }, [user?.token, handleFetchError, setSavedSearches]);
+
+  // note that selected value will be the search id since select values
+  // cannot be objects. We then need to look up the the correct search in the list
+  const handleSelectChange = useCallback(
+    (event: React.ChangeEvent<HTMLSelectElement>) => {
+      const selectedId = event?.target?.value;
+      setSelectedSavedSearch(selectedId);
+      if (selectedId) {
+        const searchToApply = savedSearches.find(
+          (search) => search.saved_search_id === selectedId,
+        )?.search_query;
+        if (!searchToApply) {
+          setApiError(new Error("Unable to find saved search"));
+          return;
+        }
+        const searchQueryParams = searchToQueryParams(searchToApply);
+        setApplyingSavedSearch(true);
+        replaceQueryParams(searchQueryParams);
+      }
+    },
+    [savedSearches, replaceQueryParams],
+  );
+
+  // fetch saved searches on page load or log in
+  // not explicitly dependent on user changes, but fetchSavedSearches is, so will still fire
+  useEffect(() => {
+    fetchSavedSearches().catch(handleFetchError);
+  }, [fetchSavedSearches, handleFetchError]);
+
+  // fetch saved searches on new saved search
+  useEffect(() => {
+    if (newSavedSearches.length && fetchSavedSearches) {
+      fetchSavedSearches()
+        .then(() => {
+          // set the latest saved search as selected
+          setSelectedSavedSearch(newSavedSearches[0]);
+        })
+        .catch(handleFetchError);
+    }
+  }, [handleFetchError, fetchSavedSearches, newSavedSearches]);
+
+  // reset saved search selector on search change
+  useEffect(() => {
```

Comment:
My reading of the docs suggests not using an effect for state changes: [*If there is no external system involved (for example, if you want to update a component’s state when some props or state change), you shouldn’t need an Effect.*](https://react.dev/learn/you-might-not-need-an-effect). It looks like the `useEffect()` could be removed.

Reply:
OK, maybe something to keep our eye on as we go. We could refactor later if we come up with a different solution.